pkg/graveler/graveler.go

```diff
@@ -1428,32 +1428,30 @@ func (g *Graveler) Set(ctx context.Context, repository *RepositoryRecord, branch
 func (g *Graveler) safeBranchWrite(ctx context.Context, log logging.Logger, repository *RepositoryRecord, branchID BranchID, stagingOperation func(branch *Branch) error) error {
 	var try int
 	for try = 0; try < BranchWriteMaxTries; try++ {
-		branch, err := g.GetBranch(ctx, repository, branchID)
+		branchPreOp, err := g.GetBranch(ctx, repository, branchID)
 		if err != nil {
 			return err
 		}
-		startToken := branch.StagingToken
-
-		if err = stagingOperation(branch); err != nil {
+		// startToken := branchPreOp.StagingToken
+		if err = stagingOperation(branchPreOp); err != nil {
 			return err
 		}
 
 		// Checking if the token has changed.
 		// If it changed, we need to write the changes to the branch's new staging token
-		branch, err = g.GetBranch(ctx, repository, branchID)
+		branchPostOp, err := g.GetBranch(ctx, repository, branchID)
 		if err != nil {
 			return err
 		}
-		endToken := branch.StagingToken
-		if startToken == endToken {
+		if branchPreOp.StagingToken == branchPostOp.StagingToken {
 			break
-		} else {
-			// we got a new token, try again
-			log.WithField("try", try+1).
-				WithField("startToken", startToken).
-				WithField("endToken", endToken).
-				Info("Retrying Set")
 		}
+		// we got a new token, try again
+		log.WithFields(logging.Fields{
+			"try":               try + 1,
+			"branch_token_pre":  branchPreOp.StagingToken,
+			"branch_token_post": branchPostOp.StagingToken,
+		}).Info("Retrying Set")
```

Comment:
This can _and will_ happen a lot under load!  I think "INFO" is much too high, it is an operation that can repeatedly log but does not log per-request.  So I'd go with "DEBUG'.
```suggestion
		}).Debug("Retrying Set")
```